app/models/user.rb

```diff
@@ -125,7 +126,11 @@ def with_all_associations_joined
     def with_interest(interest)
       where(":interest = ANY(interested_in)", interest: interest)
     end
-  end
+
+    def immutable_params
```

Comment:
As suggested before: could you name this `immutable_attributes` please? Rationale: in the context of this AR model, attributes is more appropriate. "params" hints at a controller context.